roles/openshift_health_checker/library/portavail.py

```diff
@@ -0,0 +1,47 @@
+#!/usr/bin/python
+"""Attempt to bind to a given set of ports, or return errors encountered"""
+
+import socket
+
+from ansible.module_utils.basic import AnsibleModule
+
+
+def main():
+    """Bind to a given set of ports and return any errors this might cause."""
+    module = AnsibleModule(
+        argument_spec=dict(
+            # ports is a list of dicts consisting of a port_name and a port field:
+            #   [
+            #     {
+            #       "port_name": "mysql",
+            #       "port": 3306,
+            #     }
+            #   ]
+            ports=dict(type="list", required=True),
+        ),
+    )
+
+    errors = list()
+
+    ports = module.params["ports"]
+    for port in ports:
+        if not port.get("port"):
+            continue
+
+        sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+        try:
+            sock.bind(('localhost', port["port"]))
+        except socket.error as error:
+            port["error"] = str(error)
+            errors.append(port)
+        sock.close()
```

Comment:
Perhaps this should be

```python
finally:
    sock.close()
```

This check would do worse than no check if it clogs up ports.